### PR TITLE
Refactor

### LINKED ISSUE #169: Reposition sector filter
From Rothesay,
Filter selection boxes would ideally be located in the same space – currently the filter for scenarios and sectors are half a screen apart

### PR DESCRIPTION
**Pull request by krzmip**
Created new path (active if report_version==5), which gives the same reports as version 4 but using new refactored functions.
Main changes:
- reduced server side code to essentially calling two functions (get_report_settings and produce_report)
- get_report_settings creates a list of lists containing all the required information to produce report except questionnaire inputs
- produce_report is a universal function that can produce any of the reports in any of the two formats (html, rtf)
- added several unit tests and a report test (a test that generates some reports and checks if they are unchanged)

Less important UI-related changes
- added modal message when HTML report is under construction
- restricted conditions to refresh HTML report to prevent the report from rendering again after clicking "prev" on the report tab
- getting sidebar_toc feature to a more mature state
- wrapped report page input elements in a div with max width (fix #169)

### CODE REVIEW

**Comment by krzmip** [comment]
I am a bit confused with tests. If I run devtools::test() the result is OK, but when I run devtools::check() it is failed. I will try to investigate, but do not feel expert with this. If you have any help suggestions I am open

**Comment by krzmip** [comment]
Finally clean check. Turned out that wrong capitalisation of the image name does not impact local execution but does impact github. Strange

**Review by JohnAdders — COMMENTED** [review]
Overall looks good, what' sthe plan with functions to add and functions to delete, will there be a future PR to clean up?

File: DESCRIPTION [hunk 1]
```diff
@@ -34,4 +34,5 @@ Imports:
     uuid,
     yaml,
     png,
-    markdown
+    markdown,
+    testthat
```
[the remark this answers is not in the document]
Comment (krzmip) [reply DESCRIPTION:39]: moved

**Comment by krzmip** [comment]
> Overall looks good, what' sthe plan with functions to add and functions to delete, will there be a future PR to clean up?

functions_to_delete are required for report_version <= 4. If we agree to remove those version then I remove functions_to_delete (and merge function_to_add with functions)

**Pull request merged**